RHDEVDOCS-2438: Update command changes for Pipelines 1.4 (#32483)
openshift/openshift-docs: modules/op-tkn-hub-interaction.adoc

// Module included in the following assemblies:
//
// *  cli_reference/tkn_cli/op-tkn-reference.adoc

[id="op-tkn-hub-interaction_{context}"]
= Hub interaction commands

Interact with Tekton Hub for resources such as tasks and pipelines.

== hub
Interact with hub.

.Example: Display help
[source,terminal]
----
$ tkn hub -h
----

.Example: Interact with hub API server
[source,terminal]
----
$ tkn hub --api-server https://api.hub.tekton.dev
----

[NOTE]
====
For each command examples given below, you can get the corresponding sub-commands and flags by executing `tkn hub <command> --help`.
====

== hub downgrade
Downgrade an installed resource

.Example: Downgrade an installed resource
[source,terminal]
----
$ tkn hub downgrade task \
> --context $(kubectl config current-context) \
> --kubeconfig $HOME/.kube/config \

> > --namespace $KUBECONFIG \

Reviewer: > --namespace $KUBECONFIG

should we change the var to `$NAMESPACE` ?

Author: Sure. Should I change this for all examples?
BTW, I provided the default values that are available in the tekton CLI:
```
bash-4.4 ~ $ tkn hub downgrade --help
Downgrade an installed resource

Usage:
tkn hub downgrade [flags]
tkn hub downgrade [command]

Available Commands:
  task        Downgrade an installed Task by its name to a lower version

Flags:
  -c, --context string      Name of the kubeconfig context to use (default: kubectl config current-context)
  -h, --help                help for downgrade
  -k, --kubeconfig string   Kubectl config file (default: $HOME/.kube/config)
  -n, --namespace string    Namespace to use (default: from $KUBECONFIG)
      --to string           Version of Resource

Global Flags:
      --api-server string   Hub API Server URL (default "https://api.hub.tekton.dev")

Use "tkn hub downgrade [command] --help" for more information about a command.
``` 
If `$KUBECONFIG` is not the correct default (for both OpenShift and Kubernetes), I request folks with upstream access to make this change for tekton.

Author: > > --namespace $KUBECONFIG
> 
> should we change the var to `$NAMESPACE` ?

@sm43 I have changed it to `$NAMESPACE`.

Reviewer: if user doesn't pass the flag it will take the namespace value from the kubeconfig file.
hence `from $KUBECONFIG` in the flag description

Reviewer 2: how about `-namespace mynamespace` ? for other `tkn` command we are using `mynamespace`
https://docs.openshift.com/container-platform/4.7/cli_reference/tkn_cli/op-tkn-reference.html#pipeline

Reviewer: what if we skip showing the `--context` and `--kubeconfig` flags for hub commands in examples? they are available for almost all tkn command but are not shown in examples and not needed to be passed always

Author: @VeereshAradhya - Made the changes you suggested, and some other modifications to make the document look like the other `tkn` command documentation pages.

@sm43 - Removed the `--context` and `--kubeconfig` flags and shortened the commands to a single line.

// Module included in the following assemblies:
//
// *  cli_reference/tkn_cli/op-tkn-reference.adoc

[id="op-tkn-hub-interaction_{context}"]
= Hub interaction commands

Interact with Tekton Hub for resources such as tasks and pipelines.

== hub
Interact with hub.

.Example: Display help
[source,terminal]
----
$ tkn hub -h
----

.Example: Interact with hub API server
[source,terminal]
----
$ tkn hub --api-server https://api.hub.tekton.dev
----

[NOTE]
====

> For each command examples given below, you can get the corresponding sub-commands and flags by executing `tkn hub <command> --help`.

Reviewer 3: ```suggestion
For each example, to get the corresponding sub-commands and flags, run `tkn hub <command> --help`.
```
IBM Style Guide: 
"execute verb
Use only if a simpler term, such as "run", "issue", "start", or "enter" is inappropriate."

Reviewer 3: IBM SG:
**below** adjective, adverb, preposition
Do not use to refer to an earlier software version or fix level. Use "earlier". Do not use to indicate a relative
location in a document, as in "the information below". Use "the information later in this section". Alternatively,
you can use "following" to refer to a table, to art, or to a list. Example: "The following table shows ..."

====

== hub downgrade
Downgrade an installed resource

> .Example: Downgrade an installed resource

Reviewer: Downgrade an installed task `mytask` to  version `0.2`

Reviewer 2: Downgrade an installed task `mytask` to version `newversion`

Reviewer: @VeereshAradhya I thought instead of keeping a placeholder giving an example with a value would be helpful to understand.
but if placeholder is fine. I don't have any strong opinion against it, fine with me :)
same for rest of the comments.

[source,terminal]
----

> $ tkn hub downgrade task --to olderversion -n mynamespace

Reviewer 2: `tkn hub downgrade task mytask --to oldervesion -n mynamespace`

> $ tkn hub downgrade task -n mynamespace --to <resourceVersion>

Reviewer: `$ tkn hub downgrade task mytask --to 0.3 -n mynamespace`

Reviewer 2: `$tkn hub downgrade task mytask --to <resourceVersion> -n mynamespace`

----

== hub get
Get resource manifest by its name, kind, catalog, and version.

> .Example: Get resource manifest

Reviewer: Get resource manifest of task `mytask` of version `0.2`  from `tekton` catalog

Reviewer 2: same here.. instead of giving number to version we can use `newversion`
WDYT?

[source,terminal]
----

> $ tkn hub get [pipeline | task] --from tekton --version <resourceVersion>

Reviewer: `$ tkn hub get task mytask --from tekton --version 0.2`

Reviewer 2: `$tkn hub get [task|pipeline] --from tekton --version version myresource`?

----

== hub info
Display information about a resource by its name, kind, catalog, and version.

> .Example: Display resource information

Reviewer: Display resource information of latest version of task `mytask` from `tekton` catalog

Reviewer 2: @sm43 whats wrong with using `--version`? Anyway we are using `--version` in previous commands

Author: Mentioning `mytask` and `tekton` in the description would be too specific; I think the description should be generic as it is, and the example should contain the specifics.

Reviewer: just to clarify 😅 all my comments were for examples not for command description
```
 Example: Display resource information of latest version of task `mytask` from `tekton` catalog
 ```

[source,terminal]
----

> $ tkn hub info task --from tekton --version <resourceVersion>

Reviewer: `$ tkn hub info task mytask --from tekton`

----

== hub install
Install a resource from a catalog by its kind, name, and version.

> .Example: Install a resource

Reviewer: Install latest verion of task `mytask`

Author: Same as my earlier comment - I think it's not a good idea to mention specifics in the general description. I am anyway including them in the example.

[source,terminal]
----

> $ tkn hub install task -n mynamespace --from tekton --version <resourceVersion>

Reviewer: `$ tkn hub install task mytask  --from tekton`

Author: What happens when the user wants to install a specific version of the task from tekton hub? I guess if the version is omitted, then by default the latest version available will be installed, which might not always be desired.

----

== hub reinstall
Reinstall a resource by its kind and name.

> .Example: Reinstall a resource

Reviewer: Reinstall an installed task `mytask` from `tekton` catalog with version `0.2` in `mynamespace` namespace

Author: Mentioning something too specific in the general description is not a good idea.

[source,terminal]
----

> $ tkn hub reinstall task -n mynamespace --from tekton --version <resourceVersion>

Reviewer: `$ tkn hub reinstall task mytask -n mynamespace --from tekton --version 0.2`

----

== hub search
Search a resource by a combination of name, kind, and tags.

> .Example: Search a resource

Reviewer: Search a resource by tag `cli`

Author: No specifics in the general description. There are other options apart from `tag` which can be used for search, and hence ...

[source,terminal]
----
$ tkn hub search --tags cli
----

== hub upgrade
Upgrade an installed resource.

> .Example: Upgrade an installed resource

Reviewer: Upgrade an installed task `mytask` to version `0.3`

[source,terminal]
----

> $ tkn hub upgrade task -n mynamespace --to <resourceVersion>

Reviewer: `$ tkn hub upgrade task mytask --to 0.3 -n mynamespace`
